IDEA-Prompted Cleanup Changes

voidsong-dragonfly opened the pull request:
Feel free to ask me to undo some of these, but IDEA suggested these for a variety of reasons, generally including brevity and performance (compiler can better optimize more narrow routes, etc)

voidsong-dragonfly committed:
First pass of basic IDEA-prompted code cleanup

 - Updated to J17 standards some code, simplifying it
 - Fixed some things IDEA called "probable bugs"
 - Removed redundant casts where appropriate
 - IDEA-prompted performance changes

voidsong-dragonfly committed:
Change FluidStack null checks over to isEmpty(), FS can't be null

TwistedGate commented on src/main/java/flaxbeard/immersivepetroleum/api/crafting/IPRecipeTypes.java:
```diff
@@ -23,11 +23,11 @@ public static void modConstruction() {
-				return ImmersivePetroleum.MODID+":"+name;
-			}
-		});
+		return REGISTER.register(name, () -> new RecipeType<>() {
```
No space between the two `(){`, ever

TwistedGate commented on src/main/java/flaxbeard/immersivepetroleum/api/crafting/IPRecipeTypes.java:
```diff
@@ -23,11 +23,11 @@ public static void modConstruction() {
-		});
+		return REGISTER.register(name, () -> new RecipeType<>() {
+            @Override
+            public String toString() {
```
Same for this one..

TwistedGate commented on src/main/java/flaxbeard/immersivepetroleum/api/crafting/SulfurRecoveryRecipe.java:
```diff
@@ -1,9 +1,6 @@
-import java.util.HashMap;
-import java.util.Map;
-import java.util.Objects;
+import java.util.*;
```
tbh i just hate wildcard imports

voidsong-dragonfly replied:
Okay _that_ one is on me not noticing IDEA doing that, good catch!

TwistedGate commented on src/main/java/flaxbeard/immersivepetroleum/client/gui/DerrickScreen.java:
```diff
@@ -46,9 +46,7 @@ protected void init(){
 		this.topPos = (this.height - this.imageHeight) / 2;
 		
-		this.cfgButton = new Button(this.leftPos + 125, this.topPos + 52, 50, 20, new TextComponent("Config"), button -> {
-			this.minecraft.setScreen(new DerrickSettingsScreen(this));
```
I've done it like this for reason: Readability.

voidsong-dragonfly replied:
Fair enough, I find IDEA\s way more readable but I get you

TwistedGate commented on src/main/java/flaxbeard/immersivepetroleum/client/gui/DerrickSettingsScreen.java:
```diff
@@ -43,25 +43,19 @@ protected void init(){
 		addRenderableWidget(this.pipeConfig);
 		
-		addRenderableWidget(new Button(this.guiLeft + (this.xSize / 2) - 65, this.guiTop + this.ySize - 25, 40, 20, new TextComponent("Set"), b -> {
-			MessageDerrick.sendToServer(this.derrickScreen.tile.getBlockPos(), this.pipeConfig.getGrid());
```
All of these too, readability.

TwistedGate commented on src/main/java/flaxbeard/immersivepetroleum/client/gui/ProjectorScreen.java:
```diff
@@ -108,18 +108,10 @@ protected void init(){
-		addRenderableWidget(new RotateRightButton(this.guiLeft + 115, this.guiTop + 130, but -> {
-			this.settings.rotateCW();
-		}));
+		addRenderableWidget(new CancelButton(this.guiLeft + 115, this.guiTop + 34, but -> MCUtil.getScreen().onClose()));
```
Again.. readability..

TwistedGate commented on src/main/java/flaxbeard/immersivepetroleum/api/crafting/builders/SulfurRecoveryRecipeBuilder.java:
```diff
@@ -54,9 +54,7 @@ public SulfurRecoveryRecipeBuilder addSecondaryInputFluid(TagKey<Fluid> fluid, i
 	
 	public SulfurRecoveryRecipeBuilder addItemWithChance(ItemStack item, double chance){
-		return addWriter(jsonObject -> {
-			jsonObject.add("secondary_result", this.serializerItemStackWithChance(item, chance));
```
Goodbye readability..

TwistedGate commented on src/main/java/flaxbeard/immersivepetroleum/client/gui/elements/PipeConfig.java:
```diff
@@ -116,47 +114,43 @@ public void updateTexture(){
-					case EMPTY:{
-						if((gx >= texCenterX - 2 && gx <= texCenterX + 2) && (gy >= texCenterY - 2 && gy <= texCenterY + 2)){
+
+				switch (this.grid.get(gx, gy)) {
```
As with `(){` having no space, keywords like `switch()` don't have spaces between them either.

Same with `if`, `else`, `try`, `catch` and all the others.

TwistedGate commented on src/main/java/flaxbeard/immersivepetroleum/client/particle/FluidSpill.java:
```diff
@@ -90,8 +90,7 @@ public ParticleRenderType getRenderType(){
 		public Particle createParticle(FluidParticleData type, ClientLevel world, double x, double y, double z, double xSpeed, double ySpeed, double zSpeed){
-			FluidSpill fluidSpill = new FluidSpill(type.getFluid(), world, x, y, z, xSpeed, ySpeed, zSpeed);
-			return fluidSpill;
+			return new FluidSpill(type.getFluid(), world, x, y, z, xSpeed, ySpeed, zSpeed);
```
?!

TwistedGate commented on src/main/java/flaxbeard/immersivepetroleum/client/render/debugging/DebugRenderHandler.java:
```diff
@@ -89,128 +89,125 @@ public void renderDebuggingOverlay(RenderGameOverlayEvent.Post event){
 				if(rt != null){
-					switch(rt.getType()){
-						case BLOCK:{
+					switch (rt.getType()) {
```
Formatting

TwistedGate commented on src/main/java/flaxbeard/immersivepetroleum/common/IPSaveData.java:
```diff
@@ -42,7 +43,7 @@ public IPSaveData(CompoundTag nbt){
 					
 					ImmersivePetroleum.log.info("[ReservoirIslands]: Read islands for dim {}", dimType.toString());
-					List<ReservoirIsland> list = islands.stream().map(inbt -> ReservoirIsland.readFromNBT((CompoundTag) inbt)).filter(o -> o != null).collect(Collectors.toList());
+					List<ReservoirIsland> list = islands.stream().map(inbt -> ReservoirIsland.readFromNBT((CompoundTag) inbt)).filter(Objects::nonNull).collect(Collectors.toList());
```
`Nonnull` throws a NPE, which should not happen here. Anything null is being ignored/deleted.

voidsong-dragonfly replied:
Oh good point, thanks

TwistedGate commented on src/main/java/flaxbeard/immersivepetroleum/common/cfg/ConfigUtils.java:
```diff
@@ -24,7 +24,7 @@ public static void addFuel(List<? extends String> list){
 			int index = 0;
 			
-			while(remain.indexOf(",") != -1){
+			while(remain.contains(",")){
```
under the hood, both do basicly the same thing. But whatever

TwistedGate commented on src/main/java/flaxbeard/immersivepetroleum/common/util/commands/IslandCommand.java:
```diff
@@ -22,29 +22,19 @@ private IslandCommand(){
 	}
 	
```
Wouldnt be suprised if when i merge this i loose complete sense of what the fuck is even going on Lol

Yep, you guessed it: Readablity.

TwistedGate commented on src/main/java/flaxbeard/immersivepetroleum/common/util/projector/MultiblockProjection.java:
```diff
@@ -186,18 +185,18 @@ private void updateData(){
+				case CLOCKWISE_90 -> this.offset.set(1 - size.getZ(), 0, 0);
+				case CLOCKWISE_180 -> this.offset.set(1 - size.getX(), 0, 1 - size.getZ());
+				case COUNTERCLOCKWISE_90 -> this.offset.set(0, 0, 1 - size.getX());
+				default -> this.offset.set(0, 0, 0);
```
Following the formatting or trying to restore kind of the style is not realy your strong suit is it? :D

voidsong-dragonfly replied:
I didn't realize some things were changed - do you have a stylesheet for me to import? Didn't see one but that was on me if there is one.

TwistedGate replied:
I kind of do, but i don't think it's supported by IDEA since im using Eclipse? But even *it* is not *realy* a 1:1 thing to my style. But it's as close to it as i could figure this out Lol

TwistedGate commented on src/main/java/flaxbeard/immersivepetroleum/common/util/projector/Settings.java:
```diff
@@ -193,11 +193,11 @@ public String toString(){
 	}
 	
-	public static enum Mode{
+	public enum Mode{
```
Weird how in some classes you switch to static, but the one class that get's re-created a lot gets its own thing :D

voidsong-dragonfly replied:
Enums having static is redundant, basically

TwistedGate replied:
i seee

TwistedGate reviewed (commented):
*boy* are there a lot of this different..

Loads of formatting things that i'd have to restore..

voidsong-dragonfly committed:
First batch of changes for Twisted

 - Style changes, removing excess added spaces & * imports
 - Objects::nonnull call removed
 - No readability changes yet, those coming in a moment

TwistedGate commented:
I mean generally i'd cleanup and reformat things anyway so it's whatever.. even if it takes me a whole day, mostly because i *have* to look at the code more often than other people do so why not keep it in the style im most comfortable with =P
anyway.. getting sidetracked.

Think i'll merge this tomorrow, unless you want to change/add some things to it~

voidsong-dragonfly committed:
Finish changes for Twisted

 - Change lambda readability back to before
 - Fix a few uncaught spaces

voidsong-dragonfly committed:
Change some constants to final

voidsong-dragonfly commented:
Should have finished up all the changes I want to do, unless you want me to specify all the `@Nonnull` and `@Nullable` annotations on methods that IDEA wants me to do for clarity

voidsong-dragonfly committed:
Fix up javadoc errors where possible

TwistedGate commented on src/main/java/flaxbeard/immersivepetroleum/api/crafting/reservoir/ReservoirIsland.java:
```diff
@@ -207,14 +205,14 @@ public static int getFlow(float pressure){
-	 * @return
+	 * @param level {@link Level} to query in
+	 * @param x x-coordinate to extract from
+	 * @param z z-coordinate to extract from
```
`getPressure` does not extract, but rather (as the name suggests) returns the pressure of the reservoir (if one exists)

voidsong-dragonfly replied:
Whoopsie - I must've done an accidental copy-paste error there XD

TwistedGate commented:
> Should have finished up all the changes I want to do, unless you want me to specify all the `@Nonnull` and `@Nullable` annotations on methods that IDEA wants me to do for clarity

sure

voidsong-dragonfly commented:
I'm also fixing all of your comparing of fluidstacks/itemstacks to null, if there's a null itemstack around we have a worse issue than a sudden NPE - null itemstacks went in the transition to 1.8, and null fluidstacks in the transition to 1.13.

voidsong-dragonfly committed:
Clean up code null handling & finalize some vars

 - Itemstacks & Fluidstacks are no longer compared to null, that functionality having been removed in 1.8/1.13 respectively
 - @Nullable & @Nonnull annotations are applied to every method their parent has one on
 - Useless null checks other places are removed, mostly due to stricter enforcement of annotations
 - Several things passing null when they should not have been removed
 - Make fluid tanks and other things whose object does not change final
     - This does not make their contents unable to change - only the outer object

TwistedGate commented on src/main/java/flaxbeard/immersivepetroleum/api/crafting/IPRecipeTypes.java:
```diff
@@ -18,16 +18,16 @@
+            public String toString(){
+                return ImmersivePetroleum.MODID + ":" + name;
+            }
+        });
```
Why spaces instead of tabs?

voidsong-dragonfly replied:
Think that's just what IDEA defaults to - seems like there's a mix of spaces and tabs in this project for some reason

TwistedGate replied:
Should be Tab, if there are spaces around then it's definitely a mistake Lol

voidsong-dragonfly replied:
Cool, gimme a moment then

voidsong-dragonfly replied:
Ran a replace on all four-space space areas with tabs, you should be back to full tabs now!

TwistedGate commented on src/main/java/flaxbeard/immersivepetroleum/api/crafting/reservoir/ReservoirHandler.java:
```diff
@@ -186,13 +182,13 @@ public static Reservoir addReservoir(ResourceLocation id, Reservoir reservoir){
-	public static double getValueOf(@Nonnull Level world, int x, int z){
-		if(!world.isClientSide && world instanceof WorldGenLevel worldGen){
+	public static double getValueOf(@Nonnull Level Level, int x, int z){
+		if(!Level.isClientSide && Level instanceof WorldGenLevel worldGen){
```
not sure i (nor Eclipse) like that the variable name starts with Uppercase

voidsong-dragonfly replied:
Whoopsie! I didn't even catch that, yep - I'll change it back

TwistedGate commented on src/main/java/flaxbeard/immersivepetroleum/api/crafting/reservoir/ReservoirHandler.java:
```diff
@@ -268,27 +264,19 @@ private static List<ColumnPos> keepOutline(Level world, List<ColumnPos> poly){
 						ColumnPos p = new ColumnPos(pos.x + 1, pos.z);
-						if(!set.contains(p)){
-							set.add(p);
-						}
```
*That is nessesary!*

It's supposed to *clean up*, not basicly copy everything from A to B.

voidsong-dragonfly replied:
It's a _set_ :p 
![image](https://user-images.githubusercontent.com/40431605/183068044-b87c3257-baac-49a1-9227-c642d1997ef2.png)
That's what IDEA tells me for set documentation: they can't have duplicate objects, so that check is useless, because the set _already does it for you_

voidsong-dragonfly replied:
Resolved via discord

voidsong-dragonfly committed:
Fix variable name being uppercase

voidsong-dragonfly committed:
Spaces back to tabs

voidsong-dragonfly committed:
Fix more space -> tab conversions

TwistedGate reviewed (approved):
Seems good

TwistedGate commented:
Anything else? Otherwise i'd merge it now.

voidsong-dragonfly commented:
Nothing that goes in _this_ PR - bugfixes would go in another to keep em properly organized
